### ISSUE TITLE
Issue about the padding when building the PatchEmbedding layer

### ISSUE THREAD

**Issue by what-is-available-for-name** [ViTAE-Transformer/ViTPose#125]
When I step-by-step debugged with the config (ViTPose_small_simple_coco_256x192.py), 
I found a confusing value in the parameters of the PatchEmbedding layer as the `padding` below:
`self.proj = nn.Conv2d(in_chans, embed_dim, kernel_size=patch_size, stride=(patch_size[0] // ratio), padding=4 + 2 * (ratio//2-1))`

When I run the `train.py`, the `padding `value would be `padding = 4 + 2 * (1 // 2 - 1) = 2`. It would cause the backbone to miss two columns at the right side of the origin image and two rows at the bottom of the origin image, because `Patch_size = 16` and it divides evenly into the `img_size`(256x192).

Could you please explain the details about this issue?

**Comment by what-is-available-for-name** [comment]
[1.pdf](https://github.com/ViTAE-Transformer/ViTPose/files/13785535/1.pdf)
A figure in pdf to help understand my issue
